tests/std/tests/P1206R7_vector_from_range/test.cpp

```diff
@@ -161,6 +161,11 @@ void test_vso1591034() {
     assert(counted_item::count == 6);
 }
 
+void test_LWG_3743() { // COMPILE-ONLY
+    auto r                  = ranges::subrange(views::iota(0ULL) | views::take(5), 5);

```

Comment:
This appears to be initializing `auto r` from a temporary `subrange`, which is a pattern that we conventionally avoid. It would be simpler to write this as `ranges::subrange r(INITIALIZER);` (or with braces).

(I consistently object to this case because I find this "almost always auto" pattern to be unnecessarily verbose. It's simpler to reserve `auto` for the cases where we're deducing the type of a thing from the type of another thing. When we're making a thing from scratch and we're saying the type directly on the RHS, `auto` provides no value.)